tools/run.py

```diff
@@ -0,0 +1,392 @@
+
+            nth_mixture = update_nth_mixture(nth_mixture, riridx, time_idx, event_duration_nl, sample_onoffsets, traj_doas, nl, layer, ev_traj, fold_event_filenames, is_moving, is_flipped_moving, ev_speed, event_nl, cfg)
+
+            time_idx += event_duration_nl
+
+    sort_idx = np.argsort(nth_mixture['event_onoffsets'][:,0])
+    nth_mixture['files'] = nth_mixture['files'][sort_idx]
+    nth_mixture['class'] = nth_mixture['class'][sort_idx]
+    nth_mixture['event_onoffsets'] = nth_mixture['event_onoffsets'][sort_idx]
+    #nth_mixture['sample_onoffsets'] = nth_mixture['sample_onoffsets'][sort_idx]
+    nth_mixture['trajectory'] = nth_mixture['trajectory'][sort_idx]
+    nth_mixture['isMoving'] = nth_mixture['isMoving'][sort_idx]
+    nth_mixture['isFlippedMoving'] = nth_mixture['isFlippedMoving'][sort_idx]
+    nth_mixture['speed'] = nth_mixture['speed'][sort_idx]
+    nth_mixture['rirs'] = np.array(nth_mixture['rirs'],dtype=object)
+    nth_mixture['rirs'] = nth_mixture['rirs'][sort_idx]
+    new_doas = np.zeros(len(sort_idx),dtype=object)
+    new_sample_onoffsets = np.zeros(len(sort_idx),dtype=object)
+    upd_idx = 0
+    for idx in sort_idx:
+        new_doas[upd_idx] = nth_mixture['doa_azel'][idx].T
+        new_sample_onoffsets[upd_idx] = nth_mixture['sample_onoffsets'][idx]
+        upd_idx += 1
+    nth_mixture['doa_azel'] = new_doas
+    nth_mixture['sample_onoffsets'] = new_sample_onoffsets 
+
+    return nth_mixture
+
+def main():
+    '''
+    main function to trigger the data generation
+    '''
+
+    # parse config arguments
+    args = parse_args()
+    cfg = load_config(args, args.path_to_config)
+    print('generating data using parameters:')
+    print(yaml.dump(dict(cfg), allow_unicode=True, default_flow_style=False))
+
+    
+    event_filenames = get_sound_event_filenames(cfg.PATH_TO_SOUND_EVENT_FILES)
+    event_durs = get_sound_event_filedurs(cfg.PATH_TO_SOUND_EVENT_DURS)
+
+
+    fold_names = cfg.FOLD_NAMES
+    snr_range = cfg.SNR_RANGE
+    max_polyphony = cfg.MAX_POLYPHONY
+    mixtures = []
+
+    # iterate over fold names
+    for ifold, fold in enumerate(fold_names):
+        
+        fold_event_filenames, fold_event_durs = get_fold_files_and_durs(
+                fold, 
+                event_filenames, 
+                event_durs)
+        n_mixtures_per_fold = cfg.N_MIX_PER_FOLD[ifold]
+
+        # iterate over rooms
+        fold_rooms = cfg.FOLD_ROOMS[fold]
+        room_mixtures=[]
+        for room_name in fold_rooms:
+            fold_mixture = {'mixture': []}
+            fold_mixture['roomidx'] = room_name
+
+            # get the room's relevant info
+            path_to_room_files = get_path_to_room_files(room_name)
+            room_trajs = get_room_trajectories(path_to_room_files)
+            n_traj = len(room_trajs)
+            traj_doas = get_traj_doas(room_trajs, n_traj)
```

Comment:
@ChrisIck please have a look at this strategy, and let me know if using sofa makes sense here instead. @adrianSRoman you have also dealt with files in sofa format. What do we think about this? 

In any case, we will need scripts to turn the original TAU, METU, etc. files into such file and adapt these functions to obtain these pieces of information.

Reply:
I think this _could_ make sense, but outside of the TAU-SRIR database, the split of rooms into multiple trajectories isn't really a thing, so it might make sense to format the datasets into just a single array of all RIR measurements/DOAs. We can further discuss this in our meeting tomorrow.

Reply:
Other than that, this looks good.

Reply:
Thanks for the comments Chris. 
The alternative to these pre-defined trajectories would be to create some sort of function that can look at all the RIRs in a given room and "draw" a trajectories for a specific event.

Reply:
I actually like this idea of "drawing" arbitrary trajectories out of the available RIRs in a room. Which of you would like to prototype this function?

Reply:
So #20 removes the trajectory/height structure and flattens the SRIRs in the same way the generator code does. I've pushed these, which shouldn't change how the generator works. I'll open an issue re: arbitrary trajectories and start work on that soon.